Fix docs horizontal scrollbar

patrickhlauke opened the pull request:
Addresses the scrolling issues in our docs

Note: this only deals with the scrolling of the docs caused by certain examples and the way code is presented in the docs. For the former cases, a more fundamental fix is likely still necessary to address the overflowing problem at the root.

Closes https://github.com/twbs/bootstrap/issues/24877

patrickhlauke committed:
Fix docs horizontal scrollbar in Components > Modal

patrickhlauke commented:
Before:
> ![3-components-modal](https://user-images.githubusercontent.com/895831/33242028-1f4dc4f6-d2c6-11e7-9380-8815106f5fcf.png)

After:
> ![3-components-modal-fix](https://user-images.githubusercontent.com/895831/33242023-132fe1e0-d2c6-11e7-9636-3a3c7a0ca99e.png)

patrickhlauke committed:
Add overflow to docs example class

patrickhlauke commented:
Added further fix that adds `overflow:scroll` to the examples class in the docs. This papers over all remaining issues where the docs have horizontal scrollbar (though the root causes still persist of course).

patrickhlauke commented:
Fixes look like this (horizontal overflow scrollbar)

> ![1-components-buttons-fixed](https://user-images.githubusercontent.com/895831/33243867-8e11b52c-d2e5-11e7-9d71-e28cb02bb33b.png)

> ![2-components-input-group-fixed](https://user-images.githubusercontent.com/895831/33243873-9ee92fba-d2e5-11e7-9cde-29255b252b9b.png)

> ![4-components-pagination-fixed](https://user-images.githubusercontent.com/895831/33243875-a0c61316-d2e5-11e7-8708-fcf639584a5d.png)

> ![5-utilities-text-fixed](https://user-images.githubusercontent.com/895831/33243876-a60e1918-d2e5-11e7-9b6a-a7df68577023.png)

mdo commented on assets/scss/_component-examples.scss:
```diff
@@ -118,6 +118,7 @@
   position: relative;
   padding: 1rem;
   margin: 1rem (-$grid-gutter-width / 2);
+  overflow: scroll;
```
`overflow: auto` is a better option here—it'll only show a scrollbar when necessary vs always showing one.

patrickhlauke replied:
DOH! good catch, feel stupid for using `scroll`. my CSS must be rusty. changed.

mdo commented on docs/4.0/components/modal.md:
```diff
@@ -459,7 +459,11 @@ For modals that simply appear rather than fade in to view, remove the `.fade` cl
 
-If the height of a modal changes while it is open, you should call `$('#myModal').data('bs.modal').handleUpdate()` or `$('#myModal').modal('handleUpdate')` to readjust the modal's position in case a scrollbar appears.
+If the height of a modal changes while it is open, you should call
+{% highlight js %}$('#myModal').data('bs.modal').handleUpdate(){% endhighlight %}
```
Rather than introduce inline highlighting here (which we don't do anywhere else), let's rewrite the paragraph.

> If you change the contents of a modal while it's open, you may need to manually account for changes to the modal's height.
> 
> {% highlight js %}
> // Option 1
> $('#myModal').modal('handleUpdate')
>
> // Option 2
> $('#myModal').data('bs.modal').handleUpdate()
> {% endhighlight %}

Might need better comments than `Option 1/2` though?

patrickhlauke replied:
Any hint as to when to use one option over the other? I admit that this is the part that's currently not explained at all in the docs, and I never grokked it myself (as i never had a need to dynamically alter modal content before)

patrickhlauke replied:
@mdo digging a bit further, it seems the variant with `data(...)` is related to our programmatic API, but it's just a preference (some authors may use this)? as we don't seem to do this consistently anywhere else (we seem to just stick to the documented API methods), I'm actually more inclined to drop this here (as it then avoids having to go off on a tangent on why authors may or may not choose one over the other, which is best kept to the programmatic API docs page anyway).

patrickhlauke committed:
Drop the handleUpdate() variant

While I understand the variant covers the "get a particular plugin instance" case from http://getbootstrap.com/docs/4.0/getting-started/javascript/#programmatic-api it seems weird that we do it here but not everywhere else in the docs (randomly checking other pages, we seem to stick to the documented methods as written on those pages). As this isn't properly explained as is, and as it adds nothing (more of a preference for authors), it would be easier to just drop the variant and stick with the documented method as outlined later on in the same page (avoids having to go off on a "why should you use this instead of that" tangent here).

patrickhlauke committed:
Merge branch 'v4-dev' into docs-scroll-fix

mdo reviewed (approved):
Nice, thanks! Appreciate the shorter content.

mdo commented:
Also, I agree on the component behaviors here, but I've never found a good responsive variation for these many of our components that didn't require extra code to make them work.

patrickhlauke commented:
> Nice, thanks! Appreciate the shorter content.

de nada. it scratches an itch that i had for a while (the scrolling issue, as well as "there's something in the docs that I don't have a clue about...")

> Also, I agree on the component behaviors here, but I've never found a good responsive variation for these many of our components that didn't require extra code to make them work.

I'll have a look if at least some of these can't be tackled in a reasonably simple way (perhaps with flex) soon...

patrickhlauke committed:
Merge branch 'v4-dev' into docs-scroll-fix

patrickhlauke commented:
@mdo do these issues (moved via automation etc) still need to be manually noted in the ship list?

mdo commented:
Not every change needs to be there. This one is small enough that I’m not overly concerned about noting it in the ship list. Automation only applies to the project board, not the ship list. I need to find an integration for automating that part :).
